arangod/Aql/OptimizerRulesFeature.cpp

```diff
@@ -377,6 +377,14 @@ void OptimizerRulesFeature::addRules() {
                OptimizerRule::makeFlags(OptimizerRule::Flags::CanBeDisabled,
                                         OptimizerRule::Flags::ClusterOnly));
 
+#if 0
+  // not yet enabled - we need to adjust a lot of tests in order to turn
+  // on this rule
+  registerRule("move-filters-into-enumerate", moveFiltersIntoEnumerateRule, OptimizerRule::moveFiltersIntoEnumerateCollection,
+               OptimizerRule::makeFlags(OptimizerRule::Flags::CanBeDisabled,
+                                        OptimizerRule::Flags::DisabledByDefault));
+#endif
```

Comment:
If we de-if this, it still won't be used unless explicitly enabled, right? So let's compile it in.

Reply:
No, it's not exactly equivalent. If it is compiled in but disabled by default, it will count as a "skipped" optimizer rule in the optimizer's query statistics. And that will make a test fail, which expects a certain number of rules to be skipped.

I think I will address the whole testing in a seperate PR, and until then would like to keep the rule completely inactive.